packages/libs/mcp-sdk/src/definitions.ts

```diff
@@ -0,0 +1,319 @@
+ */
+export type ParameterType = z.infer<typeof ParameterTypeEnum>;
+
+/**
+ * Mapping of parameter types to their corresponding Zod validation schemas
+ *
+ * This map provides validation logic for each supported parameter type.
+ * It is used by the buildMcpParamDefinitions function to create Zod schemas for tool parameters.
+ *
+ * @internal
+ */
+const ZodSchemaMap: Record<ParameterType, z.ZodTypeAny> = {
+  number: z.coerce.number(),
+  number_array: z.coerce.number().array(),
+  bool: z.boolean(),
+  bool_array: z.string().refine(
+    (val) =>
+      val === '' ||
+      val.split(',').every((item) => {
+        const trimmed = item.trim().toLowerCase();
+        return (
+          trimmed === '' || ['true', 'false', 'yes', 'no', '1', '0', 'y', 'n'].includes(trimmed)
+        );
+      }),
+    {
+      message: 'Must be comma-separated boolean values or empty',
+    }
+  ),
+  address: z.string().regex(/^(0x[a-fA-F0-9]{40}|0x\.\.\.|)$/, {
+    message: 'Must be a valid Ethereum address, 0x..., or empty',
+  }),
+  address_array: z.string().refine(
+    (val) =>
+      val === '' ||
+      val.split(',').every((item) => {
+        const trimmed = item.trim();
+        return trimmed === '' || trimmed === '0x...' || /^0x[a-fA-F0-9]{40}$/.test(trimmed);
+      }),
+    {
+      message: 'Must be comma-separated Ethereum addresses or empty',
+    }
+  ),
+  string: z.string(),
+  string_array: z.string(),
+  bytes: z.string(),
+  bytes_array: z.string(),
+} as const;
+
+/**
+ * Builds Zod schema definitions for Vincent tool parameters
+ *
+ * This function takes an array of Vincent parameter definitions and creates a Zod schema
+ * that can be used to validate tool inputs. Each parameter is mapped to its corresponding
+ * validation schema from the ZodSchemaMap.
+ *
+ * @param params - Array of Vincent parameter definitions
+ * @param addDelegatorPkpAddress - Whether to add the delegator eth address as a param
+ * @returns A Zod schema shape that can be used to create a validation schema
+ *
+ * @example
+ * ```typescript
+ * const parameters: VincentParameter[] = [
+ *   {
+ *     name: 'address',
+ *     type: 'address',
+ *     description: 'Ethereum address'
+ *   },
+ *   {
+ *     name: 'amount',
+ *     type: 'number',
+ *     description: 'Amount to transfer'
+ *   }
+ * ];
+ *
+ * const paramSchema = buildMcpParamDefinitions(parameters);
+ * const validationSchema = z.object(paramSchema);
+ * ```
+ */
+export function buildMcpParamDefinitions(
+  params: VincentParameter[],
+  addDelegatorPkpAddress: boolean
+): ZodRawShape {
+  const zodSchema = {} as ZodRawShape;
+
+  // Add the delegator PKP Eth address as a param. Delegatee is using the MCP and must specify which delegator to execute tools for
+  if (addDelegatorPkpAddress) {
+    zodSchema['pkpEthAddress'] = z
+      .string()
+      .describe(
+        "The delegator's PKP address that will execute the swap. For example 0xd8dA6BF26964aF9D7eEd9e03E53415D37aA96045."
```

Comment:
Description here is a bit too specific for a generic parameter

Reply:
Fixed. It must have passed by when I made this generic on all tools for delegatee mode